AppKit/CoreAnimation/CPAnimationContext.j

```diff
@@ -152,24 +152,29 @@ CPLog.debug(_cmd + "context stack =" + _CPAnimationContextStack);
 
     duration = [animation duration] || [self duration];
 
-    var needsFrameTimer = (aKeyPath == @"frame" || aKeyPath == @"frameSize") &&
-                          ([anObject hasCustomLayoutSubviews] || [anObject hasCustomDrawRect]) &&
-                          (objectId = [anObject UID]);
+    var needsPeriodicFrameUpdates = (aKeyPath === @"frame" || aKeyPath === @"frameSize" || aKeyPath === @"frameOrigin") &&
+                                    [[anObject animator] wantPeriodicFrameUpdates] &&
+                                    (objectId = [anObject UID]);
 
     if (_completionHandlerAgent)
         _completionHandlerAgent.increment();
 
     var completionFunction = function()
     {
-        if (needsFrameTimer)
+        if (needsPeriodicFrameUpdates)
+        {
             [self stopFrameUpdaterWithIdentifier:objectId];
+            [anObject _setForceUpdates:YES];
+        }
 
         if (animationCompletion)
             animationCompletion();
 
-        if (needsFrameTimer || animationCompletion)
+        if (needsPeriodicFrameUpdates || animationCompletion)
             [[CPRunLoop currentRunLoop] performSelectors];
 
+        [anObject _setForceUpdates:NO];
```

Comment:
Well, in fact there are plenty of places where it's implicit that we're dealing with views...